File: samples/client/3_0_3_unit_test/java/src/test/java/org/openapijsonschematools/client/components/schemas/ObjectTypeMatchesObjectsTest.java
package org.openapijsonschematools.client.components.schemas;

import org.junit.Assert;
import org.junit.Test;
import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;
import org.openapijsonschematools.client.schemas.validation.MapUtils;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.AbstractMap;

public class ObjectTypeMatchesObjectsTest {
    static final SchemaConfiguration configuration = new SchemaConfiguration(new JsonSchemaKeywordFlags.Builder().format().build());

    @Test
    public void testAnObjectIsAnObjectPasses() throws ValidationException {
        // an object is an object
        final var schema = ObjectTypeMatchesObjects.ObjectTypeMatchesObjects1.getInstance();
        schema.validate(
            MapUtils.makeMap(
            ),
            configuration
        );
    }

    @Test
    public void testAnArrayIsNotAnObjectFails() {
        // an array is not an object
        final var schema = ObjectTypeMatchesObjects.ObjectTypeMatchesObjects1.getInstance();
        try {
            schema.validate(
                Arrays.asList(
                ),
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }

    @Test
    public void testAnIntegerIsNotAnObjectFails() {
        // an integer is not an object
        final var schema = ObjectTypeMatchesObjects.ObjectTypeMatchesObjects1.getInstance();
        try {
            schema.validate(
                1,
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }

    @Test
    public void testABooleanIsNotAnObjectFails() {
        // a boolean is not an object
        final var schema = ObjectTypeMatchesObjects.ObjectTypeMatchesObjects1.getInstance();
        try {
            schema.validate(
                true,
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }

    @Test
    public void testAStringIsNotAnObjectFails() {
        // a string is not an object
        final var schema = ObjectTypeMatchesObjects.ObjectTypeMatchesObjects1.getInstance();
        try {
            schema.validate(
                "foo",
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }

    @Test
    public void testAFloatIsNotAnObjectFails() {
        // a float is not an object
        final var schema = ObjectTypeMatchesObjects.ObjectTypeMatchesObjects1.getInstance();
        try {
            schema.validate(
                1.1d,
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }

    @Test
    public void testNullIsNotAnObjectFails() {
        // null is not an object
        final var schema = ObjectTypeMatchesObjects.ObjectTypeMatchesObjects1.getInstance();
        try {
            schema.validate(
                (Void) null,
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }
}

File: samples/client/3_0_3_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/components/schemas/ObjectTypeMatchesObjectsTest.kt
package org.openapijsonschematools.client.components.schemas

import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertFailsWith

class ObjectTypeMatchesObjectsTest {
    companion object {
        val configuration = SchemaConfiguration(JsonSchemaKeywordFlags.Builder().format().build())
    }

    @Test
    fun testAnObjectIsAnObjectPasses() {
        // an object is an object
        val schema = ObjectTypeMatchesObjects.ObjectTypeMatchesObjects1.getInstance()
        schema.validate(
            mapOf<Any?, Any?>(
            ),
            configuration
        )
    }

    @Test
    fun testAnArrayIsNotAnObjectFails() {
        // an array is not an object
        val schema = ObjectTypeMatchesObjects.ObjectTypeMatchesObjects1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    listOf<Any?>(
                    ),
                    configuration
                )
            }
        )
    }

    @Test
    fun testAnIntegerIsNotAnObjectFails() {
        // an integer is not an object
        val schema = ObjectTypeMatchesObjects.ObjectTypeMatchesObjects1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    1,
                    configuration
                )
            }
        )
    }

    @Test
    fun testABooleanIsNotAnObjectFails() {
        // a boolean is not an object
        val schema = ObjectTypeMatchesObjects.ObjectTypeMatchesObjects1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    true,
                    configuration
                )
            }
        )
    }

    @Test
    fun testAStringIsNotAnObjectFails() {
        // a string is not an object
        val schema = ObjectTypeMatchesObjects.ObjectTypeMatchesObjects1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    "foo",
                    configuration
                )
            }
        )
    }

    @Test
    fun testAFloatIsNotAnObjectFails() {
        // a float is not an object
        val schema = ObjectTypeMatchesObjects.ObjectTypeMatchesObjects1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    1.1,
                    configuration
                )
            }
        )
    }

    @Test
    fun testNullIsNotAnObjectFails() {
        // null is not an object
        val schema = ObjectTypeMatchesObjects.ObjectTypeMatchesObjects1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    null,
                    configuration
                )
            }
        )
    }
}
